docs/telemetry.md

```diff
@@ -0,0 +1,72 @@
+# Data Layer Observer Telemetry
+
+DLO's telemetry implementation is inspired by Open Telemetry. While DLO doesn't depend on Open Telemetry, DLO telemetry can be configured to be measured by another provider like Open Telemetry or exported to a custom destination for visibility.
+
+## Exporting Telemetry to a Custom Destination
+
+By default, a `DefaultTelemetryProvider` instance is responsible for measuring telemetry (elapsed time, error counts, etc.). If the default telemetry measurement behavior is sufficient, a custom `TelemetryExporter` may be configured to send measured telemetry to a custom destination for visibility. Within this `TelemetryExporter`, telemetry events can also be transformed to the format expected by the custom destination.
+
+```
+window['_dlo_telemetryExporter']= {
+  sendSpan: (span: TelemetrySpanEvent): void => {
+    // Send span to custom telemetry destination, transforming to expected payload
+    myTelemetryDestination.post('span', {
+      name: span.name,
+      value: span.duration,
+      metadata: span.attributes,
+      timestamp: span.timestamp
+    });
+  },
+
+  sendCount: (count: TelemetryCountEvent): void => {
+    // Send count to custom telemetry destination, transforming to expected payload
+    myTelemetryDestination.post('count', {
+      name: count.name,
+      value: count.value,
+      metadata: count.attributes,
+      timestamp: count.timestamp
+    });
+  }
+}
+```
+
+## Customizing Telemetry Measurement
+
+If `DefaultTelemetryProvider` measurement behavior is insufficient, telemetry measurement and exporting can be entirely overridden by configuring a custom `TelemetryProvider`. By configuring a custom `TelemetryProvider` other telemetry libraries like Open Telemetry may be used for telemetry measurement and exporting.
+
+```
+class OpenTelemetryProvider {
```

Comment:
Interested in thoughts about including a legit Open Telemetry example here. I think it helps illustrate how to implement a custom telemetry provider with a real world target, but also has the potential to rot as the Open Telemetry API surface evolves.

Reply:
I like that you're including an OTel example.  A couple of suggestions:

1. Change the title to `## Customizing Telemetry Measurement - Open Telemetry Example`
2. Include the `import` statement that defines the types and classes in your code sample:

```
import {Tracer, Meter, SimpleSpanProcessor, ...etc} from 'open-telemetry' // I don't know what the actual package name is
```

Reply:
Maybe this? https://www.npmjs.com/package/@opentelemetry/sdk-trace-web

Reply:
Great suggestion. This forced me to reconcile changes in their metrics API surface.

Reply:
Nice! Thank you.